Apply Rubocop fixes

parkr opened the pull request:
Each cop is addressed in its own commit.

parkr committed:
Auto-correct Style/FrozenStringLiteralComment

parkr committed:
Auto-correct Layout/EmptyLineAfterMagicComment

parkr committed:
Auto-correct Style/StringLiterals

parkr committed:
Auto-correct Layout/TrailingWhitespace

parkr committed:
Auto-correct Layout/SpaceInsideHashLiteralBraces

parkr committed:
Auto-correct Layout/EmptyLinesAroundBlockBody

parkr committed:
Auto-correct Layout/SpaceAfterComma

parkr committed:
Remove empty file: spec/pagination_spec.rb

parkr committed:
Auto-correct Layout/HashAlignment

parkr committed:
Auto-correct Layout/FirstArgumentIndentation

parkr committed:
Auto-correct Layout/EmptyLineAfterGuardClause

parkr committed:
Auto-correct Style/LineEndConcatenation

parkr committed:
Auto-correct Style/RedundantException

parkr committed:
Auto-correct Layout/EmptyLinesAroundClassBody

parkr committed:
Auto-correct Layout/ClosingParenthesisIndentation

parkr committed:
Auto-correct Style/TrailingCommaInHashLiteral

parkr committed:
Add required_ruby_version to >=2.5.0

parkr committed:
Auto-correct Style/Encoding

ashmaroli commented:
@parkr Is this repo back in maintenance?
Regardless, The CI check is failing because even though `rubocop-jekyll` has been added to the bundle in a prior commit, you forgot to add a `.rubocop.yml` and load the RuboCop extension.

parkr commented:
@ashmaroli I get dependabot alerts from GitHub and the jekyll-paginate one was bugging me so that's why I updated the gemspec. Easy enough– it's a dev dependency. I added CI back in to test it and the CI workflow I copied was from jekyll-seo-tag which includes Rubocop. I am not opposed to making the rest of these style updates and merging them in, though I'm concerned we might introduce a bug by accident so the alternative is to remove the Rubocop checks from CI 😄

ashmaroli commented:
> I get dependabot alerts from GitHub and the jekyll-paginate one was bugging me so that's why I updated the gemspec...

Oh yeah, I get those too. But I just ignore them.
Since that was the sole motivation behind recent commit(s), I don't see the point in further investing time in this.
